packages/plugins-core/src/types/actualPlugin.ts

```diff
@@ -0,0 +1,521 @@
+
+export type PluginCellValue = { name: string; value: unknown | null };
+
+export interface PluginSpreadsheet {
+  /**
+   * Bind to a cell and observe changes
+   * @param sheetName - Name of the sheet (optional, defaults to global)
+   * @param binding - Cell binding (string name or object with name and optional query)
+   * @param callback - Function called when cell value changes
+   * @returns Cleanup function to stop observing
+   */
+  bind(
+    sheetName: string | undefined,
+    binding: PluginBinding,
+    callback: (node: PluginCellValue) => void,
+  ): () => void;
+
+  /**
+   * Get a cell value directly
+   * @param sheetName - Name of the sheet
+   * @param name - Cell name
+   * @returns Promise that resolves to the cell value
+   */
+  get(sheetName: string, name: string): Promise<PluginCellValue>;
+
+  /**
+   * Get all cell names in a sheet
+   * @param sheetName - Name of the sheet
+   * @returns Promise that resolves to array of cell names
+   */
+  getCellNames(sheetName: string): Promise<string[]>;
+
+  /**
+   * Create a query in a sheet
+   * @param sheetName - Name of the sheet
+   * @param name - Query name
+   * @param query - The query to create
+   * @returns Promise that resolves when query is created
+   */
+  createQuery(sheetName: string, name: string, query: Query): Promise<void>;
+}
+
+export type PluginMigration = [
+  timestamp: number,
+  name: string,
+  upCommand: string,
+  downCommand: string,
+];
+
+// Plugin context type for easier reuse
+export type PluginContext = Omit<
+  HostContext,
+  | 'registerSlotContent'
+  | 'pushModal'
+  | 'registerRoute'
+  | 'registerDashboardWidget'
+> & {
+  registerSlotContent: (
+    location: SlotLocations,
+    element: ReactElement,
+  ) => string;
+  pushModal: (element: ReactElement, modalProps?: BasicModalProps) => void;
+  registerRoute: (path: string, routeElement: ReactElement) => string;
+
+  // Dashboard widget registration - wrapped for JSX elements
+  registerDashboardWidget: (
+    widgetType: string,
+    displayName: string,
+    element: ReactElement,
+    options?: {
+      defaultWidth?: number;
+      defaultHeight?: number;
+      minWidth?: number;
+      minHeight?: number;
+    },
+  ) => string;
+
+  // Theme methods - simple and direct
+  addTheme: (
+    themeId: string,
+    displayName: string,
+    colorOverrides: ThemeColorOverrides,
+    options?: {
+      baseTheme?: 'light' | 'dark' | 'midnight';
+      description?: string;
+    },
+  ) => void;
+
+  overrideTheme: (
+    themeId: 'light' | 'dark' | 'midnight' | string,
+    colorOverrides: ThemeColorOverrides,
+  ) => void;
+
+  db?: PluginDatabase;
+  q: QueryBuilder;
+
+  // Report and spreadsheet utilities
+  createSpreadsheet: () => PluginSpreadsheet;
+
+  makeFilters: (
+    conditions: Array<PluginFilterCondition>,
+  ) => Promise<PluginFilterResult>;
+};
+
+export interface ActualPlugin {
+  name: string;
+  version: string;
+  uninstall: () => void;
```

Comment:
Can we rename to deactivate to keep consistent with `activate`?
```suggestion
  deactivate: () => void;
```